Add "Save character" button

matthew-mcallister opened the pull request:
Added a save and delete button.

ltngonnguyen reviewed (approved):
Looking great, I was planning to make a similar function but seems like your implementation is a tad more thorough. Good work!

oobabooga commented:
I like this but there are some issues:

1) Turn template only applies to instruct mode now
2) As of today, the name1/name2/context boxes for instruct and chat mode are different, so that all of them should be used as input and the appropriate boxes should be used in the function based on mode

oobabooga commented on modules/chat.py:
```diff
@@ -560,3 +561,27 @@ def upload_your_profile_picture(img, name1, name2, mode):
+    data = {k: v for k, v in data.items() if v is not None}
+
+    filepath = Path(f'{folder}/{name}.json')
+    filepath.write_text(json.dumps(data), encoding='utf-8')
```
yaml should definitely be used instead of json.

matthew-mcallister commented:
Addressed comments and also added a delete button since the lack of one annoyed me.

Character loading currently doesn't support instruct mode any longer. Should that be implemented in this PR or can it be a separate PR?

oobabooga commented:
I have opted to not include save/delete buttons for instruction templates because that seems to not be a routine task
